h5py/api_functions.txt

```diff
@@ -80,12 +80,12 @@ hdf5:
 
   # === H5D - Dataset API =====================================================
 
-  hid_t     H5Dcreate2(hid_t loc_id, char *name, hid_t type_id, hid_t space_id, hid_t lcpl_id, hid_t dcpl_id, hid_t dapl_id)
-  hid_t     H5Dcreate_anon(hid_t file_id, hid_t type_id, hid_t space_id, hid_t plist_id, hid_t dapl_id)
+  hid_t     H5Dcreate2(hid_t loc_id, char *name, hid_t type_id, hid_t space_id, hid_t lcpl_id, hid_t dcpl_id, hid_t dapl_id) nogil
+  hid_t     H5Dcreate_anon(hid_t file_id, hid_t type_id, hid_t space_id, hid_t plist_id, hid_t dapl_id) nogil
 
-  hid_t     H5Dopen(hid_t file_id, char *name)
-  hid_t     H5Dopen2(hid_t loc_id, char *name, hid_t dapl_id )
-  herr_t    H5Dclose(hid_t dset_id)
+  hid_t     H5Dopen(hid_t file_id, char *name) nogil
+  hid_t     H5Dopen2(hid_t loc_id, char *name, hid_t dapl_id ) nogil
+  herr_t    H5Dclose(hid_t dset_id) nogil
```

Comment:
I'd be inclined to keep the GIL for things like open/close, iterate, etc. They're presumably only a tiny bit of I/O before it goes back to Python code. Maybe you know more about this, though?

Reply:
It makes sense for a lot of this to be `nogil` to me, fwiw

Reply:
I had a quick look and those function are mainly error-handling and ref-counting. It is probably not worth releasing the GIL for them.